osquery/tables/system/darwin/certificate_trust_settings.cpp

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <iomanip>
+
+#include <CoreFoundation/CoreFoundation.h>
+#include <Security/Security.h>
+
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/sql/sql.h>
+#include <osquery/tables/system/darwin/keychain.h>
+#include <osquery/utils/conversions/darwin/cfdata.h>
+#include <osquery/utils/conversions/darwin/cfnumber.h>
+#include <osquery/utils/conversions/darwin/cfstring.h>
+
+#define kSecTrustSettingsPolicyName CFSTR("kSecTrustSettingsPolicyName")
+
+namespace osquery {
+namespace tables {
+
+const std::map<std::string, SecTrustSettingsDomain> kSecTrustSettingsDomains = {
+    {"admin", kSecTrustSettingsDomainAdmin},
+    {"system", kSecTrustSettingsDomainSystem},
+    {"user", kSecTrustSettingsDomainUser},
+};
+
+void getCertificateTrustSettingsForDomain(std::string domain_name,
+                                          SecTrustSettingsDomain domain,
+                                          QueryData& results) {
+  CFArrayRef certificates;
+  OSStatus status = SecTrustSettingsCopyCertificates(domain, &certificates);
+  if (status != errSecSuccess) {
+    CFStringRef error = SecCopyErrorMessageString(status, nil);
+    TLOG << "Failed to copy certificate trust settings for domain: " << domain
+         << ". Error: " << error;
+    CFRelease(error);
+    return;
+  }
+
+  CFIndex cert_count = CFArrayGetCount(certificates);
+  for (CFIndex i = 0; i < cert_count; i++) {
+    Row r;
+    r["trust_domain"] = domain_name;
+    OSStatus cert_status;
+    SecCertificateRef cert =
+        (SecCertificateRef)CFArrayGetValueAtIndex(certificates, i);
+
+    CFStringRef common_name = nullptr;
+    cert_status = SecCertificateCopyCommonName(cert, &common_name);
+    if (common_name) {
+      if (cert_status == errSecSuccess) {
+        r["common_name"] = stringFromCFString(common_name);
+      }
+      CFRelease(common_name);
+    }
+
+    CFDataRef serial_number = SecCertificateCopySerialNumberData(cert, nil);
+    if (serial_number) {
+      const UInt8* serial_bytes = CFDataGetBytePtr(serial_number);
+      CFIndex serial_length = CFDataGetLength(serial_number);
+      std::stringstream ss;
+      ss << std::hex << std::setfill('0');
+
+      for (CFIndex o = 0; o < serial_length; o++) {
+        ss << std::setw(2) << static_cast<int>(serial_bytes[o]);
+      }
+
+      r["serial_number"] = ss.str();
```

Comment:
If we generate this as an uppercase string and rename the column to `serial`, we are consistent with the `certificates` table.